src/main/java/at/hannibal2/skyhanni/features/misc/TabWidgetSettings.kt

```diff
@@ -0,0 +1,99 @@
+package at.hannibal2.skyhanni.features.misc
+
+import at.hannibal2.skyhanni.SkyHanniMod
+import at.hannibal2.skyhanni.events.GuiContainerEvent
+import at.hannibal2.skyhanni.events.InventoryCloseEvent
+import at.hannibal2.skyhanni.events.InventoryFullyOpenedEvent
+import at.hannibal2.skyhanni.utils.ItemUtils.getLore
+import at.hannibal2.skyhanni.utils.LorenzColor
+import at.hannibal2.skyhanni.utils.LorenzUtils
+import at.hannibal2.skyhanni.utils.RenderUtils.highlight
+import at.hannibal2.skyhanni.utils.StringUtils.anyMatches
+import at.hannibal2.skyhanni.utils.StringUtils.matches
+import at.hannibal2.skyhanni.utils.repopatterns.RepoPattern
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+
+class TabWidgetSettings {
+    private val patternGroup = RepoPattern.group("tab.widget.setting")
+    private val mainPageSettingPattern by patternGroup.pattern(
+        "gui",
+        "WIDGETS.*"
+    )
+    private val mainPageWidgetPattern by patternGroup.pattern(
+        "main",
+        "§7Currently:.*"
+    )
+    private val subPageWidgetPattern by patternGroup.pattern(
+        "sub",
+        "§eClick to .*"
+    )
+    private val shownSettingPattern by patternGroup.pattern(
+        "show",
+        "Shown .* Setting.* |.*Widget Settings"
+    )
+    private val clickToDisablePattern by patternGroup.pattern(
+        "click.disable",
+        ".*(disable!)"
+    )
+    private val enabledPattern by patternGroup.pattern(
+        "is.enabled",
+        ".*(ENABLED)"
+    )
+
+    var inInventory = false;
+    var highlights = mutableMapOf<Int, LorenzColor>()
+
+    @SubscribeEvent
+    fun onInventoryOpen(event: InventoryFullyOpenedEvent) {
+        if (!isEnabled()) return
+        highlights.clear()
+
+        val inventoryName = event.inventoryName
+        if (mainPageSettingPattern.matches(inventoryName.uppercase())) {
+            val items = event.inventoryItems.filter { mainPageWidgetPattern.anyMatches(it.value.getLore()) }
+            for ((slot, stack) in items) {
+                highlights[slot] = if (enabledPattern.anyMatches(stack.getLore())) {
+                    LorenzColor.GREEN
+                } else {
+                    LorenzColor.RED
+                }
+            }
+        }
+
+        if (shownSettingPattern.matches(inventoryName)) {
+            val items = event.inventoryItems.filter {
+                subPageWidgetPattern.matches(it.value.getLore().lastOrNull())
+            }
+
+            for ((slot, stack) in items) {
+                if (subPageWidgetPattern.matches(stack.getLore().lastOrNull())) {
```

Comment:
This `if` here is unnecessary. We filter the same thing already a few lines above. Either remove the `if` or the filter.